### DISCUSSION TITLE
RTL support

### DISCUSSION THREAD

**Discussion by ilia4bit** [marp-team/marp#238] [Q&A] [answered]
Hey thanks for your amazing project it's very useful ❤️
i want create presentation with Persian/Farsi language but i'm new to marp.
**is marp support rtl direction by default ?**
can i do something like this in vscode ?
```markdown
---
marp: true
direction: rtl
---
```
**or should i use marpit framework to create custom theme which supports rtl dircetion ?**

**Comment by yhatt** [comment, marked as answer]
We have not yet full support for RTL direction, but thank you for your feedback about RTL!

For now, you can try to set [`direction` CSS property](https://developer.mozilla.org/en-US/docs/Web/CSS/direction) into each slides `section`.

```markdown
---
marp: true
style: |
  section {
    direction: rtl;
  }
---
```

Note that Marp and built-in themes still have many things of not considering RTL direction, like [CSS Logical Properties](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Logical_Properties), slide navigation in the exported HTML, and more. We're welcome to contributions for full RTL support in Marp!

**Reply by BSeyfi** [reply]
@yhatt  RTL is a need and we appreciate any improvement. Thanks you for the MARP
